hub/api/dataset.py

```diff
@@ -53,7 +53,7 @@ class Dataset:
     def __init__(
         self,
         url: str,
-        mode: str = "a",
+        mode: str = "r+",
```

Comment:
We don't have an "r+" mode, it will be treated as "r".

Reply:
which mode is read and write? is it "rb" without files being overwritten?

Reply:
Check this PR https://github.com/activeloopai/Hub/issues/318

Reply:
there is "r", "a", "w" for now. We don't have "r+".

Reply:
@edogrigqv2 we need to have read and write access without overwriting.